tempus_fugit_minecraft/window.py

```diff
@@ -407,3 +413,15 @@ def draw_reticle(self) -> None:
         """Draw the crosshairs in the center of the screen."""
         glColor3d(0, 0, 0)
         self.reticle.draw(GL_LINES)
+    '''!Updates the environments lights. Solves issue#8
+        @param dt   the amount of time that has elapsed since the last update to environment lights
+    '''
+    def update_day_night(self, dt: float) -> None:
+        self.game_time = self.game_time + 1
+        hour = math.fmod(self.game_time, 24)
+        increase_decrease_value = .2
+        if hour < 12:
+            self.shaders.decrease_light_intensity(increase_decrease_value)
+        else:
+            self.shaders.increase_light_intensity(increase_decrease_value)
+        return dt
```

Comment:
First, I feel like we could leverage the already existing update function and update the clock there, plus we'd be able to stop the movement of time with the pause mechanic already there. Since the update function is set to be called every `1 / TICKS_PER_SECOND` we could calculate the hour by:
```py
real_seconds_to_hours = 5
ticks_per_game_day = 24 * (real_seconds_to_hours * (1 / TICKS_PER_SECOND))
light_intensity_per_tick = 0.2 / (real_seconds_to_hours * (1 / TICKS_PER_SECOND))
self.game_time += 1
hour = math.fmod(self.game_time, ticks_per_game_day)
self.shaders.increase_light_intensity(light_intensity_per_tick if hour >= 12 else -light_intensity_per_tick)
```

Second, while writing the above code snippet, I just realized that 5 seconds is way too short for an in game hour since that means that the game day length is currently only 2 minutes. Lets make the number of seconds per in game hour to be 50 seconds. That gets us 1200 seconds per day, or 20 minutes per day.

Third, ideally the notion of time in game should exist in the model since this could be leveraged for other game concepts. Can we think of a way to encapsulate this in the model?

Reply:
The 5 seconds is modifiable. It's a placeholder value mostly to show off the actual effects of changing the shading. As for the second clock, it's there to keep it separated from the clock that's being used for the update function. If we had to compare it, one's an event clock and the other is a system clock. 

As for the time change calculations, that seems to be more a stylistic change. If we were to update that it would be more towards another issue such as "more realistic day night changes." 
 
The commented out stuff should be deleted.